black-jack-redo.rb

```diff
@@ -25,88 +27,212 @@ def print_with_stars(string)
+			hand_value += 10
+		when 'Q'
+			user_hand = user_hand + 'Queen of '
+			hand_value += 10
+		when 'K'
+			user_hand = user_hand + 'King of '
+			hand_value += 10
 		else
+			user_hand = user_hand + "#{current_card} of "
+			hand_value += current_card.to_i
 		end
-		count3 += 1
+
+		user_hand = user_hand + get_suit(current_hand_array[array_index])
 	end
-	return count4
+
+	if is_player == 'true'
+		user_hand = user_hand + "Your cards add up to #{hand_value} points!" + "\n"
+	else
+		user_hand = user_hand + "The dealers cards add up to #{hand_value} points!" + "\n"
+	end
+	return user_hand, hand_value
 end
 
-def dealer(count)
-	number = rand(4)
-	number = 21 - number
-	if count == number
-		return "You and the dealer both had #{number}, it's a tie!"
-	elsif count < number
-		return "The dealer had #{number} and you had #{count}, you lose!"
-	else count > number
-		return "The dealer had #{number} and you had #{count}, you win!"
+def get_value_of_hand(current_hand)
+	user_hand, hand_value = viewhand(current_hand, 'true')
+	return hand_value
+end
+
+def compare_player_dealer_points(player_points, dealer_points)
+	if player_points == dealer_points
+		winner = "You and the dealer both had #{dealer_points}, dealer wins!"
+	elsif player_points < dealer_points
+		winner = "The dealer had #{dealer_points} and you had #{player_points}, you lose!"
+	elsif player_points > dealer_points
+		winner = "The dealer had #{dealer_points} and you had #{player_points}, you win!"
+	else
+		puts "Something went wrong when determining who won!!"
 	end
+
+	return winner
 end
 
-def draw(array)
-	number = array.length - 1
-	puts array[number]
-	return array[number]
+def who_wins(current_deck, current_hand, dealer_hand)
+	user_hand, hand_value = viewhand(current_hand, 'true')
+	updated_dealer_hand, dealer_value = viewhand(dealer_hand, 'false')
+	if (dealer_value < 17) && (dealer_value < hand_value)
+		updated_dealer_hand, dealer_value = viewhand(dealer_hand, 'false')
+		print_with_stars(updated_dealer_hand)
+		while (dealer_value < 17) && (dealer_value < hand_value)
+			updated_dealer_hand, bool = drawone_card(current_deck, dealer_hand)
+			current_deck.pop
+			updated_dealer_hand, dealer_value = viewhand(dealer_hand, 'false')
+			print_with_stars(updated_dealer_hand)
+			if bool == true
+				who_wins_string = 'The dealer bust! You win!'
+				print_with_stars(who_wins_string)
+				return
+			end
+		end
+
+		who_wins_string = user_hand + updated_dealer_hand + compare_player_dealer_points(hand_value, dealer_value)
+	else
+		who_wins_string = user_hand + updated_dealer_hand + compare_player_dealer_points(hand_value, dealer_value)
+	end
+
+	print_with_stars(who_wins_string)
 end
 
-def drawone(array, cards, count)
-	bust = false
-	number = rand(52)
-	if array[number][:dealt] == true
-		draw(array, cards)
+def hand_contains_ace(current_hand)
+	if (current_hand.include?('SA')) || (current_hand.include?('HA')) || (current_hand.include?('DA') )|| (current_hand.include?('CA'))
```

Comment:
Instead of running four evaluations joined by or (`||`) operators, why not just see if the array of four aces is included in the current hand? Some things you might google:
- ruby compare two arrays
- ruby check array for multiple values
- ruby how to determine if at least one item is in array

Reply:
came up with this solution => if current_hand.length > (current_hand - ['SA','HA','DA','CA']).length